core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -123,6 +128,7 @@ public FileSourceConfig(String referenceName, @Nullable String fileRegex, @Nulla
     this.timeTable = timeTable;
     this.inputFormatClass = inputFormatClass == null ?
       CombinePathTrackingInputFormat.class.getName() : inputFormatClass;
+    this.fileSourceFormat = fileSourceFormat == null ? "text" : fileSourceFormat;
```

Comment:
you need to add some validation logic, to make sure the given format is a valid one.